src/workingtitle-vcockpits-instruments-navsystems-g3000/html_ui/WorkingTitle/Pages/VCockpit/Instruments/NavSystems/AS3000/AS3000_Common.js

```diff
@@ -157,22 +157,51 @@ class AS3000_MapElement extends MapInstrumentElement {
 			}
 		}
 		
-		if (this.lastOrientation != orientation) {
-			switch (orientation % 3) {
+		if (this.orientation != orientation) {
+			switch (orientation) {
 			case 0:
```

Comment:
I still think it might be useful to add constants in MapInstrument for the various orientations.   Feels on that?

Reply:
If we want MapInstrument to define the constants, it would make most sense for MapInstrument to define rotation behavior as well (i.e. how the code used to work). Otherwise we have the possibility of outside code using the same constants but then assigning completely different rotation behavior to them, which seems messy to me. I'm not entirely opposed to having MapInstrument handle the orientation definitions again. I'm curious what other people think. Should all the orientation handling live within MapInstrument?

Reply:
No, I think you're mostly right... tere's really no worry about individual logic using its own values if it's the only thing that's dealing with rotation.

The *one* thing that I'm slightly hung up on would be communicating status via LVar, since that can only be bool/number.  If we wanted to share the orientation mode across instruments for some reason we'd have to have an agreement on that the number represents.

But I might be overthinking / prematurely optimizing this.  I tend to do that sometimes.